src/datachain/lib/arrow.py

```diff
@@ -125,3 +128,15 @@ def _arrow_type_mapper(col_type: pa.DataType) -> type:  # noqa: PLR0911
     if isinstance(col_type, pa.lib.DictionaryType):
         return _arrow_type_mapper(col_type.value_type)  # type: ignore[return-value]
     raise TypeError(f"{col_type!r} datatypes not supported")
+
+
+def _nrows_file(file: File, nrows: int) -> str:
+    tf = NamedTemporaryFile(delete=False)
+    with file.open(mode="r") as reader:
+        with open(tf.name, "a") as writer:
+            for row, line in enumerate(reader):
+                writer.write(line)
+                writer.write("\n")
+                if row >= nrows:
```

Comment:
just to double check: will it support headers?

Reply:
Good catch. I forgot to support no-header use cases. Fixed in the latest commit by adding a row if there is a header.